lib/routes/slack.js

```diff
@@ -0,0 +1,95 @@
+module.exports = function(crowi, app) {
+  'use strict';
+
+  var slackHelper = require('../util/slack')(crowi)
+  , Page = crowi.model('Page')
+  , User = crowi.model('User')
+  , actions = {};
+
+  var api = actions.api = {};
+
+  function fetchPage(link) {
+  }
+
+  api.handleEvent = function(req, res) {
+    if (req.body.type != null && req.body.type == 'url_verification') {
+      verifyChallenge(req, res);
+      return;
+    }
+
+    let event = req.body.event;
+    switch (event.type) {
+      case 'link_shared':
+      unfurl(req, res);
+      break;
+      default:
+      break;
+    }
+  }
+
+  function verifyChallenge(req, res) {
+    res.send(req.body.challenge);
+  }
+
+  function unfurl(req, res) {
+    const https = require('https');
+    const HOST = 'slack.com'
+    const PATH = '/api/chat.unfurl';
+
+    const link = req.body.event.links[0];
+    const channel = req.body.event.channel;
+    const url = require('url').parse(link.url, true);
+
+    var pagePath = url.path || null;
+    var pageId = url.query.page_id || null; // TODO: handling
+    var revisionId = url.query.revision_id || null;
+
+    User.findAllUsers()
+    .then (function(users) {
+      return Page.findPage(pagePath, users[0], revisionId);
```

Comment:
And also the url that is not the wiki page like `/admin/xxxx` would be passed in case.
In that case, 404 will be returned and no problem, but, a bit it is unexpected behavior.
If we check the path strictly, check `isCreatableName()` before call `findPage()`
https://github.com/crowi/crowi/blob/master/lib/models/page.js#L378